storage/innobase/xtrabackup/src/backup_mysql.cc

```diff
@@ -2137,13 +2137,21 @@ void mdl_lock_tables() {
       if (fsp_is_ibd_tablespace(atoi(row[1]))) {
         char full_table_name[MAX_FULL_NAME_LEN + 1];
         innobase_format_name(full_table_name, sizeof(full_table_name), row[0]);
+        if (check_if_skip_table(row[0]) &&
+            !fsp_is_dd_tablespace(atoi(row[1]))) {
```

Comment:
We always copy mysql.ibd, so even if the tables are from mysql.ibd will be excluded by `check_if_skip_table` we want to make sure we run MDL on mysql.ibd tables.

Regarding compression dictionary, they will be excluded in the if statement that was `is_access_blocked`

Reply:
Got it. I guess it is un-necessary to do SELECT on mysql.* tables because DDLs on DD tables or to say system tables is not allowed by users. but no harm either. Ok keep to the current behaviour as is